gh-124452: Fix header mismatches when folding/unfolding with email message (#125919)
python/cpython: Misc/NEWS.d/next/Library/2024-10-24-10-49-47.gh-issue-124452.eqTRgx.rst

> Fix email parsing bug that introduced spurious leading whitespace into header values when the header includes a newline character after the ';' and before the value.

Reviewer: Maybe you should used rst label.
```
:mod:`email`
```
For more to see https://devguide.python.org/documentation/markup/

Author: true, I'll have look at this doc

> Fix :mod:`email` parsing bug that introduced spurious leading whitespace into header values when the header includes a newline character after the ';' and before the value.

Reviewer 2: ```suggestion
Fix an issue in :meth:`email.policy.EmailPolicy.header_source_parse` and
:meth:`email.policy.Compat32.header_source_parse` that introduced spurious
leading whitespaces into header values when the header includes a newline
character after the header name delimiter (`:`) and before the value.
```

Let's try to render it a bit better and say which methods were fixed.

> Fix the mismatched email headers after converting them to bytes and parsing them into an email message.
> This issue only occurs with long text without any spaces.
> Root cause: The email library doesn't know how to wrap long text without spaces, but it wraps it anyway with a leading WSP, which is not removed when converting bytes to EmailMessage.

Reviewer 3: News entries don't need to be all that long, let's keep it short:
```suggestion
Fix the mismatched :mod:`email` headers after converting them to :class:`bytes` and parsing them into an email message.
```

Reviewer 4: The actual bug is that the parser was not stripping all of the whitespace from the front of the value.  This but has nothing to do with wrapping, and would have manifested on parsing any message that had a header that started with a newline.  Such messages are rarely if ever encountered in the wild, though, so the (separate) folding bug is what caused this bug to manifest.

All of which is to say I would make the news item something like "Fix email parsing bug that introduced spurious leading whitespace into header values when the header includes a newline character after the ';' and before the value.
